runtime/drivers/repo.go

```diff
@@ -32,8 +32,12 @@ type WatchCallback func(event []WatchEvent)
 
 type WatchEvent struct {
 	Type runtimev1.FileEvent
+	// Path relative to the repo root
 	Path string
-	Dir  bool
+	// FullPath used during flush
+	FullPath string
+	Dir      bool
+	IsCreate bool
 }
```

Comment:
Feels like this is leaking implementation details in the interface. Consider using an internal wrapper instead.